src/finbars/finbars.ts

```diff
@@ -0,0 +1,870 @@
+     *
+     * @see {@link FinBar#barStyles|barStyles}
+     * @type {finbarStyles}
+     * @memberOf FinBar.prototype
+     */
+    set style(styles) {
+        var keys = Object.keys(styles = extend({}, styles, this._auxStyles));
+
+        if (keys.length) {
+            var bar = this.bar,
+                barRect = bar.getBoundingClientRect(),
+                container = this.container || bar.parentElement,
+                containerRect = container.getBoundingClientRect(),
+                oh = this.oh;
+
+            // Before applying new styles, revert all styles to values inherited from stylesheets
+            bar.setAttribute('style', BAR_STYLE);
+
+            keys.forEach(function (key) {
+                var val = styles[key];
+
+                if (key in oh) {
+                    key = oh[key];
+                }
+
+                if (!isNaN(Number(val))) {
+                    val = (val || 0) + 'px';
+                } else if (/%$/.test(val)) {
+                    // When bar size given as percentage of container, if bar has margins, restate size in pixels less margins.
+                    // (If left as percentage, CSS's calculation will not exclude margins.)
+                    var oriented = axis[key],
+                        margins = barRect[oriented.marginLeading] + barRect[oriented.marginTrailing];
+                    if (margins) {
+                        val = parseInt(val, 10) / 100 * containerRect[oriented.size] - margins + 'px';
+                    }
+                }
+
+                bar.style[key] = val;
+            });
+        }
+    },
+
+    /**
+     * @readonly
+     * @name paging
+     * @summary Enable page up/dn clicks.
+     * @desc Set by the constructor. See the similarly named property in the {@link finbarOptions} object.
+     *
+     * If truthy, listen for clicks in page-up and page-down regions of scrollbar.
+     *
+     * If an object, call `.paging.up()` on page-up clicks and `.paging.down()` will be called on page-down clicks.
+     *
+     * Changing the truthiness of this value after instantiation currently has no effect.
+     * @type {boolean|object}
+     * @memberOf FinBar.prototype
+     */
+    paging: true,
+
+    /**
+     * @name range
+     * @summary Setter for the minimum and maximum scroll values.
+     * @desc Set by the constructor. These values are the limits for {@link FooBar#index|index}.
+     *
+     * The setter accepts an object with exactly two numeric properties: `.min` which must be less than `.max`. The values are extracted and the object is discarded.
+     *
+     * The getter returns a new object with `.min` and '.max`.
+     *
+     * @type {rangeType}
+     * @memberOf FinBar.prototype
+     */
+    set range(range) {
+        validRange(range);
+        this._min = range.min;
+        this._max = range.max;
+        this.contentSize = range.max - range.min + 1;
```

Comment:
Note, this caused problems, because the content size is not +1, but the virtual rows visible in the grid.